### PR TITLE
New Rack Sprite

### PR DESCRIPTION
**Pull request by PacifistDalek**
## What Does This PR Do
Replaces our old rack sprite with a new one [from Baystation.](https://github.com/Baystation12/Baystation12/pull/32289).

EDIT: You can now place items anywhere on a rack instead of just the center

## Why It's Good For The Game
The old rack is completely unclear in what it's supposed to be. Though to be honest, I'm not 100% sure if this new one is the direction we want to go in. I figured I'd put it up here to see people's general reponse, and if it's liked, great! If it's generally not, then i'll close the PR.

## Images of changes
image

image

## Testing
Looked at it in game.

## Changelog
:cl: AzlanOnPC, PacifistDalek
imageadd: New sprites for racks.
tweak: You can now place items on any part of a rack, not just the center.
/:cl:

### CODE REVIEW

**Comment by Carthusia** [comment]
Is it possible to make racks act similar to tables in the sense that wherever you click, it places the item? Right now racks just default the item to the center from what i recall

**Comment by PacifistDalek** [comment]
> Is it possible to make racks act similar to tables in the sense that wherever you click, it places the item? Right now racks just default the item to the center from what i recall

oooh, that’d be a very nice addition to this to make them look nice

**Comment by PacifistDalek** [comment]
> Is it possible to make racks act similar to tables in the sense that wherever you click, it places the item? Right now racks just default the item to the center from what i recall

done!!

**Comment by Carthusia** [comment]
wooo!

File: code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -850,14 +850,26 @@
 	if(O.loc != src.loc)
 		step(O, get_dir(O, src))
 
+/obj/structure/rack/proc/item_placed(item)
```
Comment (S34NW) [inline code/game/objects/structures/tables_racks.dm:853]: Do we need this? It's also present on tables but only to be overriden by trays to track tray contents.

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -850,14 +850,26 @@
...
 		if(user.drop_item())
 			W.Move(loc)
+			var/list/click_params = params2list(params)
+			//Center the icon where the user clicked.
+			if(!click_params || !click_params["icon-x"] || !click_params["icon-y"])
+				return
+			//Clamp it so that the icon never moves more than 16 pixels in either direction (thus leaving the table turf)
+			W.pixel_x = clamp(text2num(click_params["icon-x"]) - 16, -(world.icon_size/2), world.icon_size/2)
+			W.pixel_y = clamp(text2num(click_params["icon-y"]) - 16, -(world.icon_size/2), world.icon_size/2)
+			item_placed(W)
```
Comment (S34NW) [inline code/game/objects/structures/tables_racks.dm:872]: and this

**Review by S34NW — APPROVED** [review]

**Comment by Denthamos** [comment]
Could we get a picture of the racks in a more open area surrounding more sprites of other things like computers, tables, ect.

**Comment by PacifistDalek** [comment]
So the more I look at these, the less I like them. Racks desparately need a resprite, but this...I don't think that's it. Anyone who thinks otherwise is more than welcome to take what i've done here and reopen it!

**Pull request closed without merging**